src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonParser.cs

```diff
@@ -0,0 +1,476 @@
+                TryMatch(token, "true", JsonKind.TrueLiteralToken, out newKind) ||
+                TryMatch(token, "null", JsonKind.NullLiteralToken, out newKind) ||
+                TryMatch(token, "false", JsonKind.FalseLiteralToken, out newKind) ||
+                TryMatch(token, "Infinity", JsonKind.InfinityLiteralToken, out newKind) ||
+                TryMatch(token, "undefined", JsonKind.UndefinedLiteralToken, out newKind))
+            {
+                return new JsonLiteralNode(token.With(kind: newKind));
+            }
+
+            if (Matches(token, "-Infinity"))
+            {
+                SplitLiteral(token, out var minusToken, out var newLiteralToken);
+
+                return new JsonNegativeLiteralNode(
+                    minusToken, newLiteralToken.With(kind: JsonKind.InfinityLiteralToken));
+            }
+
+            var firstChar = token.VirtualChars[0];
+            if (firstChar == '-' || firstChar == '.' || IsDigit(firstChar))
+            {
+                return ParseNumber(token);
+            }
+
+            return new JsonTextNode(
+                token.With(kind: JsonKind.TextToken).AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                    string.Format(WorkspacesResources._0_unexpected, firstChar.Char),
+                    firstChar.Span)));
+        }
+
+        private JsonConstructorNode ParseConstructor(JsonToken token)
+        {
+            var newKeyword = token.With(kind: JsonKind.NewKeyword);
+            var nameToken = ConsumeToken(JsonKind.TextToken, WorkspacesResources.Name_expected);
+            var openParen = ConsumeToken(JsonKind.OpenParenToken, _openParenExpected);
+
+            var savedInConstructor = _inConstructor;
+            _inConstructor = true;
+
+            var result = new JsonConstructorNode(
+                newKeyword,
+                nameToken,
+                openParen,
+                ParseSequence(),
+                ConsumeToken(JsonKind.CloseParenToken, _closeParenExpected));
+
+            _inConstructor = savedInConstructor;
+            return result;
+        }
+
+        private bool TryMatch(JsonToken token, string val, JsonKind kind, out JsonKind newKind)
+        {
+            if (Matches(token, val))
+            {
+                newKind = kind;
+                return true;
+            }
+
+            newKind = default;
+            return false;
+        }
+
+        private bool Matches(JsonToken token, string val)
+        {
+            var chars = token.VirtualChars;
+            if (chars.Length != val.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0, n = val.Length; i < n; i++)
+            {
+                if (chars[i].Char != val[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsDigit(char ch)
```

Comment:
static?
Also applies to a few methods below (ParseLIteral, ParseNumber, ParseEmptyValue).

Reply:
done.  #resolved.